README.md

```diff
@@ -86,20 +84,124 @@ code because of all the basic column-renamings and the full listing of health co
+The Environmental data was pretty straightforward as DAP provided us with a proto-schema which we
+converted to a repo schema. We decided to break the larger table up into smaller fragments: 
+_geocoding, census, pollutants, temperature_precipitation, and walkability_ variables.
+
+* A single `environment` table which contains a single Environment record per dog per month per address. 
+* The table includes 147 environmental variables and a foreign key to the dog table. 
+* Rows in the final Environment table will be unique on _dog_id_ and _address_month_year_.
+
+
+## Sample Data
+Sample Data in the Dog Aging Project is populated as different cohorts of participants are sent different 
+sample kits. Our extraction of this data seeks to capture the linkage between study_id (_dog_id_) and 
+sample_id as well as some other metadata about the sample.
+
+#### Sample Extraction
+* Pull all records from the baseline arm (*baseline_arm_1*) 
+* _k1_tube_serial_ and _k1_rtn_tracking_date_ must be populated
 
+#### Sample Schema Design
+* A single `sample` table which contains multiple samples per dog.
+* Simple lookup table with 6 fields and a foreign key to the dog table.
+* Rows in the final Environment table will be unique on _dog_id_ and _sample_id_.
+
+
+## Label Mapping Table
+We have a manually maintained lookup table that we are constructing using the Terra JSON schema for the DAP tables and 
+the DAP data dictionaries provided from RedCap. The table contains a row for each field in the the final dog aging tables.
+The mapping table includes fields to capture information about the upstream source of data, datatypes, and most importantly,
+the mapping of raw values ot value labels for RedCap survey questions with *radio* responses.
+
+We are looking forward to building some tooling that will be able to
+parse a proto-schema provided by DAP into usable code for ingest in the form of a Terra JSON Schema output.
+
+
+#### Creating Outfiles 
+DAP's long term future would include an official TDR<->Workspace integration but we are currently still building that out.
+To give ourselves some breathing room, we've written a [script](./hack/convert-output-to-tsv.py) to convert
+the outputs of our transformation pipeline into workspace-compatible TSVs. 
 The script takes two positional arguments:
 1. Path to the top-level directory containing the outputs of the transformation pipeline
 2. Path to a directory where converted TSVs should be written
+There are additional arguments to that script to add some flexibility to the tool:
+* Can take in a list of tables to process (currently looks for all tables)
+* Can output primary keys with a modified column name for Firecloud compatibility: *entity:{table_name}_id*
+* Call Format:
+    * `python convert-output-to-tsv.py {transform GS path} {tsv outfile local path} -d`
+    * `python convert-output-to-tsv.py gs://broad-dsp-monster/weekly_refresh/20210525/transform /Users/DATA/DAP/tsv -d`
```

Comment:
⛏️ I like to avoid including "real" infrastructural references if possible:
```suggestion
    * `python convert-output-to-tsv.py gs://example-bucket/weekly_refresh/20210525/transform /Users/DATA/DAP/tsv -d`
```